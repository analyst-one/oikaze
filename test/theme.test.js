/**
 * @jest-environment jest-environment-node-single-context
 */

const sass = require("sass");

it("renders css for single theme", () => {
  const { css } = sass.compile("test/fixtures/single-theme.scss");
  expect(css).toMatchSnapshot();
});

it("renders css for theme overrides", () => {
  const { css } = sass.compile("test/fixtures/multi-theme.scss");
  expect(css).toMatchSnapshot();
});

it("renders css with prefixes", () => {
  const { css } = sass.compile("test/fixtures/with-prefix.scss");
  expect(css).toMatchSnapshot();
});
