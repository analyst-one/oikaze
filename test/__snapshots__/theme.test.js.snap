// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders css for single theme 1`] = `
":root {
  --color-primary-200: #aed051;
  --color-primary-200--rgb: "174,208,81";
  --color-primary-300: #c3dd7b;
  --color-primary-300--rgb: "195,221,123";
  --color-primary-400: #d7e9a6;
  --color-primary-400--rgb: "215,233,166";
  --color-primary-600: #75912b;
  --color-primary-600--rgb: "117,145,43";
  --color-primary-500: #93b733;
  --color-primary-500--rgb: "147,183,51";
  --color-primary-700: #576b21;
  --color-primary-700--rgb: "87,107,33";
  --color-primary-800: #3a4717;
  --color-primary-800--rgb: "58,71,23";
  --color-neutral-600: #ffffff;
  --color-neutral-600--rgb: "255,255,255";
  --color-neutral-500: #ffffff;
  --color-neutral-500--rgb: "255,255,255";
  --color-neutral-400: #ffffff;
  --color-neutral-400--rgb: "255,255,255";
  --color-neutral-300: #fefefe;
  --color-neutral-300--rgb: "254,254,254";
  --color-neutral-200: #fefefe;
  --color-neutral-200--rgb: "254,254,254";
  --color-neutral-100: #fefefe;
  --color-neutral-100--rgb: "254,254,254";
  --color-neutral-800: #7f7f7f;
  --color-neutral-800--rgb: "127,127,127";
  --color-neutral-900: #000000;
  --color-neutral-900--rgb: "0,0,0";
  --color-primary: #93b733;
  --color-primary--rgb: "147,183,51";
  --color-neutral: #fefefe;
  --color-neutral--rgb: "254,254,254";
  --size-root: 16px;
  --size-root--rem: 1rem;
  --size-base: 16px;
  --size-base--rem: 1rem;
  --size-none: 0px;
  --size-none--rem: 0rem;
  --size-xx-small: 2px;
  --size-xx-small--rem: 0.125rem;
  --size-x-small: 4px;
  --size-x-small--rem: 0.25rem;
  --size-small: 8px;
  --size-small--rem: 0.5rem;
  --size-regular: 16px;
  --size-regular--rem: 1rem;
  --size-large: 32px;
  --size-large--rem: 2rem;
  --size-x-large: 64px;
  --size-x-large--rem: 4rem;
  --size-xx-large: 128px;
  --size-xx-large--rem: 8rem;
}

.test {
  color: #93b733;
  color: #93b733;
  color: #93b733;
  color: var(--color-primary, #93b733);
  color: rgba(147, 183, 51, 0.2);
  color: rgba(var(--color-primary--rgb, 147,183,51), 0.2);
  color: rgba(147, 183, 51, 0.2);
  font-size: 16px;
  font-size: var(--size-regular, 16px);
  font-size: 16px;
  font-size: var(--size-small--rem, 0.5rem);
  font-size: 0.5rem;
}"
`;

exports[`renders css for theme overrides 1`] = `
":root {
  --color-primary-200: #aed051;
  --color-primary-200--rgb: "174,208,81";
  --color-primary-300: #c3dd7b;
  --color-primary-300--rgb: "195,221,123";
  --color-primary-400: #d7e9a6;
  --color-primary-400--rgb: "215,233,166";
  --color-primary-600: #75912b;
  --color-primary-600--rgb: "117,145,43";
  --color-primary-500: #93b733;
  --color-primary-500--rgb: "147,183,51";
  --color-primary-700: #576b21;
  --color-primary-700--rgb: "87,107,33";
  --color-primary-800: #3a4717;
  --color-primary-800--rgb: "58,71,23";
  --color-neutral-600: #ffffff;
  --color-neutral-600--rgb: "255,255,255";
  --color-neutral-500: #ffffff;
  --color-neutral-500--rgb: "255,255,255";
  --color-neutral-400: #ffffff;
  --color-neutral-400--rgb: "255,255,255";
  --color-neutral-300: #fefefe;
  --color-neutral-300--rgb: "254,254,254";
  --color-neutral-200: #fefefe;
  --color-neutral-200--rgb: "254,254,254";
  --color-neutral-100: #fefefe;
  --color-neutral-100--rgb: "254,254,254";
  --color-neutral-800: #7f7f7f;
  --color-neutral-800--rgb: "127,127,127";
  --color-neutral-900: #000000;
  --color-neutral-900--rgb: "0,0,0";
  --color-primary: #93b733;
  --color-primary--rgb: "147,183,51";
  --color-neutral: #fefefe;
  --color-neutral--rgb: "254,254,254";
  --size-root: 16px;
  --size-root--rem: 1rem;
  --size-base: 16px;
  --size-base--rem: 1rem;
  --size-none: 0px;
  --size-none--rem: 0rem;
  --size-xx-small: 2px;
  --size-xx-small--rem: 0.125rem;
  --size-x-small: 4px;
  --size-x-small--rem: 0.25rem;
  --size-small: 8px;
  --size-small--rem: 0.5rem;
  --size-regular: 16px;
  --size-regular--rem: 1rem;
  --size-large: 32px;
  --size-large--rem: 2rem;
  --size-x-large: 64px;
  --size-x-large--rem: 4rem;
  --size-xx-large: 128px;
  --size-xx-large--rem: 8rem;
}

body.alt {
  --color-primary: #ff0000;
  --color-primary--rgb: "255,0,0";
  --size-root: 14px;
  --size-small: 7px;
  --size-regular: 14px;
}

.test {
  color: #ff0000;
  color: #ff0000;
  color: var(--color-primary, #ff0000);
  color: rgba(var(--color-primary--rgb, 255,0,0), 0.2);
  color: rgba(255, 0, 0, 0.2);
  font-size: var(--size-small--rem, 7px);
  font-size: 7px;
}"
`;

exports[`renders css with prefixes 1`] = `
":root {
  --ok--color-primary-200: #aed051;
  --ok--color-primary-200--rgb: "174,208,81";
  --ok--color-primary-300: #c3dd7b;
  --ok--color-primary-300--rgb: "195,221,123";
  --ok--color-primary-400: #d7e9a6;
  --ok--color-primary-400--rgb: "215,233,166";
  --ok--color-primary-600: #75912b;
  --ok--color-primary-600--rgb: "117,145,43";
  --ok--color-primary-500: #93b733;
  --ok--color-primary-500--rgb: "147,183,51";
  --ok--color-primary-700: #576b21;
  --ok--color-primary-700--rgb: "87,107,33";
  --ok--color-primary-800: #3a4717;
  --ok--color-primary-800--rgb: "58,71,23";
  --ok--color-neutral-600: #ffffff;
  --ok--color-neutral-600--rgb: "255,255,255";
  --ok--color-neutral-500: #ffffff;
  --ok--color-neutral-500--rgb: "255,255,255";
  --ok--color-neutral-400: #ffffff;
  --ok--color-neutral-400--rgb: "255,255,255";
  --ok--color-neutral-300: #fefefe;
  --ok--color-neutral-300--rgb: "254,254,254";
  --ok--color-neutral-200: #fefefe;
  --ok--color-neutral-200--rgb: "254,254,254";
  --ok--color-neutral-100: #fefefe;
  --ok--color-neutral-100--rgb: "254,254,254";
  --ok--color-neutral-800: #7f7f7f;
  --ok--color-neutral-800--rgb: "127,127,127";
  --ok--color-neutral-900: #000000;
  --ok--color-neutral-900--rgb: "0,0,0";
  --ok--color-primary: #93b733;
  --ok--color-primary--rgb: "147,183,51";
  --ok--color-neutral: #fefefe;
  --ok--color-neutral--rgb: "254,254,254";
  --ok--size-root: 16px;
  --ok--size-root--rem: 1rem;
  --ok--size-base: 16px;
  --ok--size-base--rem: 1rem;
  --ok--size-none: 0px;
  --ok--size-none--rem: 0rem;
  --ok--size-xx-small: 2px;
  --ok--size-xx-small--rem: 0.125rem;
  --ok--size-x-small: 4px;
  --ok--size-x-small--rem: 0.25rem;
  --ok--size-small: 8px;
  --ok--size-small--rem: 0.5rem;
  --ok--size-regular: 16px;
  --ok--size-regular--rem: 1rem;
  --ok--size-large: 32px;
  --ok--size-large--rem: 2rem;
  --ok--size-x-large: 64px;
  --ok--size-x-large--rem: 4rem;
  --ok--size-xx-large: 128px;
  --ok--size-xx-large--rem: 8rem;
}

.test {
  color: #93b733;
  color: #93b733;
  color: #93b733;
  color: var(--ok--color-primary, #93b733);
  color: rgba(147, 183, 51, 0.2);
  color: rgba(var(--ok--color-primary--rgb, 147,183,51), 0.2);
  color: rgba(147, 183, 51, 0.2);
  font-size: 16px;
  font-size: var(--ok--size-small--rem, 0.5rem);
  font-size: 0.5rem;
}"
`;
